src/tests/Heatmap.test.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { render } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import fetchMock from 'jest-fetch-mock';
+import { renderHook } from '@testing-library/react-hooks';
+import useFetchPosts from '../hooks/useFetchPosts';
+import App from '../App';
+import * as kittens1 from './responses/1.json';
+import * as kittens2 from './responses/2.json';
+import * as kittens3 from './responses/3.json';
+import * as kittens4 from './responses/4.json';
+import * as kittens5 from './responses/5.json';
+
+fetchMock.enableMocks();
+
+const renderSearchPage = (route) => render(
+  <MemoryRouter initialEntries={[route]}>
+    <App />
+  </MemoryRouter>,
+);
+
+describe('Heatmap', () => {
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  it('to contain post counts for each hour of each day', async () => {
+    fetch
+      .once(JSON.stringify(kittens1))
+      .once(JSON.stringify(kittens2))
+      .once(JSON.stringify(kittens3))
+      .once(JSON.stringify(kittens4))
+      .once(JSON.stringify(kittens5));
+    const { waitForNextUpdate } = renderHook(() => useFetchPosts('kittens'));
+
+    await waitForNextUpdate();
+    await act(async () => {
```

Comment:
I guess there are a couple of smaller problems. 
1. For the test of the search page these three lines are not necessary. They are only required for testing a custom hook, but not when you want to test a component or page
2. Maybe you did that already: You can use the debug function that is returned from `render` to see the DOM. So when you have a test like this you will see no heatmap table as you mentioned.
  ```
  fetch.once...
  const { getByRole, debug } = renderSearchPage('/search/kittens');
  debug()
  ```
  That's because the (mock) fetching is not done yet. During the initial render the heatmap is not visible.
3. Instead of the `getByRole` query you can use `findByRole`. This one is async, so you have to `await` it. When you run the test now you should see the `act` warnings plus a different error: `MutationObserver is not a constructor`
4. This leads us to this [comment](https://github.com/testing-library/dom-testing-library/issues/477) where you can find the solution. When you add `debug()` now after `findByRole` you should see the table in the output

Reply:
That worked!  Thanks

Reply:
Oddly, the tests are passing locally, but not remotely even though they should be using the same data, right?  Am I missing something?

Reply:
Perfect, glad that it worked :+1: 

The problem with the failing tests in the CI is probably that the GitHub pipelines have a different timezone than your local machine. You need to set a fixed timezone. You can check the cypress scripts inside package.json. There's an environment variable TZ that you can use